src/kernels/MIOpenLayerNorm.cpp

```diff
@@ -0,0 +1,259 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifdef MIOPEN_BETA_API
+
+#define GET(buf, idx) buf[idx]
+#define SET(buf, idx, val) buf[idx] = val
+
+#define GET_VAL(x, idx) GET(x, idx)
+#define SET_VAL(x, idx, val) SET(x, idx, val)
+
+#if MIOPEN_USE_FP64 == 1
+#define FSTYPE double
+#define MIOPEN_TYPE double
+#elif MIOPEN_USE_FP32 == 1
+#define FSTYPE float
+#define MIOPEN_TYPE float
+#elif MIOPEN_USE_FP16 == 1
+#define FSTYPE half
+#define MIOPEN_TYPE half
+#elif MIOPEN_USE_BF16 == 1
+#define FSTYPE hip_bfloat16
+#define MIOPEN_TYPE hip_bfloat16
+#endif
+
+#define LOCAL_SIZE 256
```

Comment:
I suggest to pass it from the host code during compilation, like you pass the other defines like `USE_FP32`.
It can help to avoid any discrepancies between host and kernel defines.